Expand helm requirements page to explain matching the helm policy

mallardduck opened the pull request.

l-mb reviewed (approved):
This change makes sense to me. While it's desirable to stay compatible with Helm versions, customers should really be encouraged to keep them reasonably uptodate as well.

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
 
 <DeprecationHelm2 />
 
+## Identifying the proper Helm 3 version
```
```suggestion
## Identifying the Proper Helm v3 Version
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
 
+## Identifying the proper Helm 3 version
+
+Select any Helm 3 version that is officially compatible with the Kubernetes version range you are using from our supported list.
```
```suggestion
Select any Helm v3 version that is officially compatible with the Kubernetes version range you are using from our [Rancher Support Matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions).
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+
+To apply this rule, you may need to reference two external resources:
+- **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.
+- **Rancher's Kubernetes Support Range:** Use the [Rancher Support Matrix](https://rancher.com/support-maintenance-terms/) to identify the Kubernetes versions supported by your target Rancher minor version.
```
```suggestion
- **Rancher's Kubernetes Support Range:** Use the [Rancher Support Matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions) to identify the Kubernetes versions supported by your target Rancher minor version.
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+- **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.
+- **Rancher's Kubernetes Support Range:** Use the [Rancher Support Matrix](https://rancher.com/support-maintenance-terms/) to identify the Kubernetes versions supported by your target Rancher minor version.
+
+### Example:
+- **Scenario:** You are targeting Rancher 2.11.4, which supports Kubernetes versions 1.30 through 1.32
```
```suggestion
### Example

- **Scenario:** You are targeting Rancher v2.11.4, which supports Kubernetes versions 1.30 through 1.32.
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+- **Scenario:** You are targeting Rancher 2.11.4, which supports Kubernetes versions 1.30 through 1.32
+- **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes 1.32.
+- **Result:** You find that both Helm 3.17 and Helm 3.18 support the Kubernetes 1.30-1.32 range.
+- Both will work, however selecting 3.18 is preferred because it is the newest Helm minor with overlapping support range.
```
```suggestion
- Although both work, we recommend Helm v3.18 because it is the newest Helm minor version overlapping the supported Kubernetes range.
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+### Example:
+- **Scenario:** You are targeting Rancher 2.11.4, which supports Kubernetes versions 1.30 through 1.32
+- **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes 1.32.
+- **Result:** You find that both Helm 3.17 and Helm 3.18 support the Kubernetes 1.30-1.32 range.
```
```suggestion
- **Result:** You find that both Helm v3.17 and Helm v3.18 support the Kubernetes v1.30-v1.32 range.
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+
+### Example:
+- **Scenario:** You are targeting Rancher 2.11.4, which supports Kubernetes versions 1.30 through 1.32
+- **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes 1.32.
```
```suggestion
- **Application:** Our rule requires a Helm version that supports this range. You can verify this by checking the Helm version's compatibility with the highest version in the range, Kubernetes v1.32.
```

pmkovar commented:
Thanks for the PR. 

Can you copy the changes to versions >=2.8 under https://github.com/rancher/rancher-docs/tree/main/versioned_docs as well?

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
-- Helm v2.16.0 or higher is required for Kubernetes v1.16. For the default Kubernetes version, refer to the [release notes](https://github.com/rancher/rke/releases) for the version of RKE that you are using.
-- Helm v2.15.0 should not be used, because of an issue with converting/comparing numbers.
-- Helm v2.12.0 should not be used, because of an issue with `cert-manager`.
+- Helm v2 support was removed in Rancher v2.9.z
```
```suggestion
- Helm v2 support was removed in Rancher v2.9.x.
```

LucasSaintarbor commented on docs/getting-started/installation-and-upgrade/resources/helm-version-requirements.md:
```diff
@@ -12,7 +12,22 @@ This section contains the requirements for Helm, which is the tool used to insta
+
+Select any Helm 3 version that is officially compatible with the Kubernetes version range you are using from our supported list.
+
+To apply this rule, you may need to reference two external resources:
+- **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.
```
```suggestion
To apply this rule, you may need to reference two external resources:

- **Helm Version Compatibility:** Refer to the [Helm Version Support Policy](https://helm.sh/docs/topics/version_skew/) and select the version matching the rule for your Rancher minor target.
```

LucasSaintarbor reviewed (changes requested):
@mallardduck Thanks for the PR! Requesting some small changes and please update this page in `versioned_docs/version-2.12/getting-started/installation-and-upgrade/resources`

mallardduck commented:
Thank you for the feedback @LucasSaintarbor - I've accepted all changes and updated the versioned docs. Per our slack chat, I also updated the 2.9 and 2.10 examples to reflect versions that makes sense for those Rancher minors.

For the 2.12/main docs I think it's OK to leave 2.11 as a static example - as it makes sense chronologically at least compared to if I left 2.11 examples in older docs.

LucasSaintarbor reviewed (approved):
@mallardduck Thanks for updating the example per version and porting the changes to the 2.9 -2.12 docs. LGTM.
